packages/x-charts-pro/src/context/ZoomProvider/Zoom.types.ts

```diff
@@ -0,0 +1,71 @@
+import { AxisId } from '@mui/x-charts/internals';
+
+export type ZoomOptions = {
+  /**
+   * The starting percentage of the zoom range. In the range of 0 to 100.
+   *
+   * @default 0
+   */
+  start?: number;
+  /**
+   * The ending percentage of the zoom range. In the range of 0 to 100.
+   *
+   * @default 100
+   */
+  end?: number;
+  /**
+   * The step size of the zooming function. Defines the granularity of the zoom.
+   *
+   * @default 5
+   */
+  step?: number;
+  /**
+   * Restricts the minimum span size in the range of 0 to 100.
+   *
+   * If the span size is smaller than the minSpan, the span will be resized to the minSpan.
+   *
+   * @default 10
+   */
+  minSpan?: number;
+  /**
+   * Restricts the maximum span size in the range of 0 to 100.
+   *
+   * If the span size is larger than the maxSpan, the span will be resized to the maxSpan.
+   *
+   * @default 100
+   */
+  maxSpan?: number;
+  /**
+   * Set to `false` to disable panning. Useful when you want to pan programmatically,
+   * or to show only a specific section of the chart.
+   *
+   * @default true
+   */
+  panning?: boolean;
```

Comment:
We might want to make this more complex. Eg: `panOnDrag|zoomOnWheel|zoomOnPinch: bool | 'shift' | 'ctrl' | 'alt'` similar to what echarts does. https://echarts.apache.org/en/option.html#dataZoom-inside.moveOnMouseMove